extensions/hibernate-search-orm-elasticsearch/deployment/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/HibernateSearchElasticsearchCdiProcessor.java

```diff
@@ -18,12 +18,14 @@
 import io.quarkus.hibernate.orm.PersistenceUnit;
 import io.quarkus.hibernate.orm.runtime.PersistenceUnitUtil;
 import io.quarkus.hibernate.search.orm.elasticsearch.runtime.HibernateSearchElasticsearchRecorder;
+import io.quarkus.hibernate.search.orm.elasticsearch.runtime.HibernateSearchElasticsearchRuntimeConfig;
 
 public class HibernateSearchElasticsearchCdiProcessor {
 
-    @Record(ExecutionTime.STATIC_INIT)
+    @Record(ExecutionTime.RUNTIME_INIT)
```

Comment:
This means we are moving some work to the runtime? Is it really what we want? And if so, is the work we move noticeable from a startup speed point of view?

Reply:
The work we're moving to runtime is the creation of two `Supplier<>` instances, one for `SearchMapping` and one for `SearchSession`.

It's negligible performance-wise, and in any case it makes more sense, because the `SearchMapping` and `SearchSession` are only available at runtime, not during static init.

Reply:
Also, the reason I did this is to have access to the runtime config when we retrieve the SearchMapping and SearchSession, which is only possible if this method is annotated with `@Record(ExecutionTime.RUNTIME_INIT)`, for some reason.